pkg/updatestrategy/node_pool_manager.go

```diff
@@ -278,97 +278,135 @@ func (m *KubernetesNodePoolManager) drain(ctx context.Context, node *Node) error
 		return err
 	}
 
-	// evictAll is a function that tries to evict all evictable pods from a particular node exactly
-	// once in order of appearance. If it encounters errors due to pod disruption budget violation it
-	// ignores this pod and continues with the next. The function returns any error encountered,
-	// including the most recent error produced by a pod disruption budget violation.
-	evictAll := func() error {
-		pods, err := m.getPodsByNode(node.Name)
-		if err != nil {
-			return err
-		}
+	pods, err := m.getPodsByNode(node.Name)
+	if err != nil {
+		return err
+	}
 
-		var lastPDBViolationErr error
-		for _, pod := range pods.Items {
-			// we check at the start because there's a continue in the loop body
-			if err = ctx.Err(); err != nil {
-				return backoff.Permanent(err)
-			}
+	errors := make(chan error, len(pods.Items))
+	for _, pod := range pods.Items {
```

Comment:
Can we extract it somewhere? The exact same logic is in node pool creation I think, at this point I think it makes sense to have something like `waitFor(funcs []func() error) []error`. Alternatively, we could just use `ErrGroup` and log the errors in the individual functions, which has the benefit of not having to implement anything.

Reply:
Changed it to use `errgroup`.